homeassistant/components/overkiz/__init__.py

```diff
@@ -43,17 +54,58 @@ class HomeAssistantOverkizData:
     scenarios: list[Scenario]
 
 
+async def async_migrate_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Migrate an old config entry."""
+
+    LOGGER.debug("Migrating from version %s", entry.version)
+
+    # v1 -> v2: CONF_HUB renamed to CONF_SERVER and CONF_API_TYPE added
+    if entry.version == 1:
+        v2_entry_data = {**entry.data}
+        v2_entry_data[CONF_SERVER] = entry.data[CONF_HUB]
+        v2_entry_data.pop(CONF_HUB)
+        v2_entry_data[CONF_API_TYPE] = APIType.CLOUD  # V1 only supports cloud
+
+        entry.version = 2
+        hass.config_entries.async_update_entry(entry, data=v2_entry_data)
+
+        LOGGER.debug("Migration to version %s successful", entry.version)
+
```

Comment:
Ideally we don't do a migration unless there is no other option.

A migration means it makes it harder for users to downgrade

Reply:
Are there ways to write a "v2 to v1" migration? Without this migration, the integration will be less clear, but there would be ways technically to work around it.

Reply:
I was just thinking about this, but the thing is, this code is run when you start HA since this is in the new version. If someone then downgrades, the version the user is going to should have that code, and it doesn't. So, yes, technically you could create a migration script, but that would only work for people who are migrating from that version to the new version, not for the rest.

Reply:
Fair points, I will see if I can work around this. Less nice data names are better than broken installations due to downgrade. 😉

Reply:
Fixed.